test/lsp.py

```diff
@@ -8,13 +8,77 @@
 import sys
 import traceback
 import re
-
+import tty
+import functools
+from collections import namedtuple
+import queue
+import threading
+from copy import deepcopy
 from typing import Any, List, Optional, Tuple, Union
-from enum import Enum, auto
+from itertools import islice
 
 import colorama # Enables the use of SGR & CUP terminal VT sequences on Windows.
 from deepdiff import DeepDiff
 
+TestRegexesTuple = namedtuple("TestRegexesTuple", [
+    "sendRequest",
+    "findQuotedTag",
+    "findTag",
+    "fileDiagnostics",
+    "diagnostic"
+])
+TEST_REGEXES = TestRegexesTuple(
+    re.compile(R'^// -> (?P<method>[\w\/]+) {'),
+    re.compile(R'(?P<tag>"@\w+")'),
+    re.compile(R'(?P<tag>@\w+)'),
+    re.compile(R'// (?P<testname>\w+):[ ]?(?P<diagnostics>[\w @]*)'),
+    re.compile(R'(?P<tag>@\w+) (?P<code>\d\d\d\d)')
+)
+
+TagRegexesTuple = namedtuple("TagRegexestuple", ["simpleRange", "multilineRange"])
+TAG_REGEXES = TagRegexesTuple(
+    re.compile(R"(?P<range>[\^]+) (?P<tag>@\w+)"),
+    re.compile(R"\^(?P<delimiter>[()]{1,2}) (?P<tag>@\w+)$")
+)
+
+
+def countIndex(sequence, start=0):
```

Comment:
```suggestion
def countIndex(sequence: YourTypeHere, start=0):
```
I'd also prefer a little documentation :)